.github/workflows/build.yml

```diff
@@ -1,85 +1,175 @@
+              -H 'accept: application/json' \
+            | jq --raw-output '.[0].version_data.major')
+          echo "LATEST_JAVA_VERSION=${LATEST_JAVA_VERSION}" >> $GITHUB_ENV
+      - name: Latest Java version
+        run: |
+          echo "Latest Java version: ${{ env.LATEST_JAVA_VERSION }}"
+    outputs:
+      version: ${{ env.LATEST_JAVA_VERSION }}
+
+  test-multiple-os:
+    name: Test in ${{ matrix.os }}, Java ${{ matrix.java-version }}
+    needs: latest-java-version
+    if: needs.latest-java-version.result == 'success' && needs.latest-java-version.outputs.version != null
     strategy:
       fail-fast: false
       matrix:
-        # Alternative for testing on multiple platforms. Ubuntu runs the quickest with the latest JVM.
-        #   os: [ubuntu-latest, windows-latest, macos-latest]
-        #   java-version: [11, 17]
-        os: [ ubuntu-latest ]
-        java-version: [ 17 ]
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+        java-version:
+          - 8
+          - 11
+          - ${{ needs.latest-java-version.outputs.version }}
     runs-on: ${{ matrix.os }}
     timeout-minutes: 10
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-        with:
-          fetch-depth: 0 # Required depth for JReleaser
       - name: Setup Java JDK
         uses: actions/setup-java@v3
         with:
-          distribution: adopt-hotspot
+          distribution: temurin
           java-version: ${{ matrix.java-version }}
-
+          check-latest: true
       # The project version extract NEEDS to have the maven wrapper already downloaded.
       # So we have a dummy step here just to initialize it.
-      - name: Download maven wrapper
+      - name: Download Maven wrapper
         run: ./mvnw --version
+      - name: Build and test
+        run: ./mvnw test
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: unit-test-${{ matrix.os }}-and-java-${{ matrix.java-version }}
+          retention-days: 21
+          path: |
+            **/TEST-*
+            **/hs_err_pid*
 
+  publish-test-results:
+    name: Publish the tests results
+    needs: test-multiple-os
+    if: always()
+    runs-on: ubuntu-latest
+    permissions:
+      checks: write
+      pull-requests: write
+    steps:
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          path: artifacts
+      - name: Publish test results
+        uses: EnricoMi/publish-unit-test-result-action@v2
+        with:
+          check_name: Unit Test results
+          files: artifacts/unit-test-*/**/*.xml
+      - name: Display files
+        run: ls -lha .
+      - name: Display structure of downloaded files
+        run: ls -lhaR artifacts
```

Comment:
I think these `ls` commands can be removed or commented out now that the CI goals have been accomplished

Reply:
Yes, I forgot to delete these steps. Removed in commit [7be7c67](https://github.com/Col-E/InstrumentationServer/pull/11/commits/7be7c678bc75becb037fab4edcaf1430c8d4b9f5)